os/pm/pm_procfs.c

```diff
@@ -447,43 +460,119 @@ static size_t power_devices_read(FAR struct file *filep, FAR char *buffer, size_
 }
 
 /****************************************************************************
- * Name: power_lock_write
+* 	Name: power_lock_write
+*   
+* 	Description:
+*	Lock a desired state by increasing the counter of the state.
+*
+* 	Input Parameters:
+*   filep             - File path of the power domain
+*   buffer            - Desired state to be unlocked
+*   buflen (NOT USED) - Length of the buffer
+*
+* 	Returned Value:
+*   0 (OK) means the state lock was successful.
+*	1 (Failed) means the input state was incorrect.
  ****************************************************************************/
 static size_t power_lock_write(FAR struct file *filep, FAR const char *buffer, size_t buflen)
 {
 	int state = atoi(buffer);
+	(void)buflen;
 
 	if (state >= PM_NORMAL && state < PM_SLEEP) {
-		pm_stay(0, state);
+		pm_stay(PM_IDLE_DOMAIN, state);
 		fvdbg("power locked!\n");
 	} else {
+		return 1;
```

Comment:
Is it right to return 1 when invalid state is passed as input to this function?

Reply:
It depends to our design, since our return value has 0/1, from the upper layer, we can add something like
`if (ret > 0) {....}` which means it has failed due to wrong input.

Reply:
Let's follow the standard TizenRT practice of returning <0 for erroneous cases here please?